record Dolby Vision 8.4 with the HEVC encoder

Smali changes of vendor_oplus_camera 9799ca1, over baksmali --sequential-labels output.

diff --git b/classes7/mm/DV.smali b/classes7/mm/DV.smali
new file mode 100644
index 0000000000000000000000000000000000000000..11ad85297aded46d2957ab512706d40fae34a95c
--- /dev/null
+++ b/classes7/mm/DV.smali
@@ -0,0 +1,728 @@
+.class public final Lmm/DV;
+.super Ljava/lang/Object;
+.source "DV.java"
+
+
+# static fields
+.field static final RPU:[B
+
+
+# direct methods
+.method static constructor <clinit>()V
+    .locals 1
+
+    .line 13
+    const/16 v0, 0x193
+
+    new-array v0, v0, [B
+
+    fill-array-data v0, :array_a
+
+    sput-object v0, Lmm/DV;->RPU:[B
+
+    return-void
+
+    :array_a
+    .array-data 1
+        0x0t
+        0x0t
+        0x0t
+        0x1t
+        0x7ct
+        0x1t
+        0x19t
+        0x8t
+        0x9t
+        0x8t
+        0x40t
+        0x61t
+        0x30t
+        0x0t
+        0x10t
+        0x20t
+        0x65t
+        0x6t
+        -0x1at
+        0xft
+        -0x3ct
+        0x53t
+        -0x67t
+        0x0t
+        0x40t
+        0x2t
+        0x50t
+        0x76t
+        0x0t
+        0x7ft
+        -0x10t
+        0x3t
+        0x13t
+        -0x17t
+        -0x7t
+        -0x44t
+        0x7ct
+        0x5dt
+        -0x2dt
+        0x19t
+        -0x36t
+        0x13t
+        0x17t
+        -0x4at
+        -0x61t
+        -0x1et
+        -0x11t
+        0x5t
+        0x3ct
+        -0x70t
+        -0x50t
+        -0x52t
+        -0x17t
+        -0x4dt
+        -0x16t
+        0x50t
+        0x6at
+        0x5ft
+        0x45t
+        -0x2at
+        -0x7ft
+        -0xft
+        -0x78t
+        0x66t
+        -0x56t
+        0x5t
+        -0x6et
+        -0x47t
+        -0x8t
+        0x45t
+        -0x76t
+        0x2et
+        -0x48t
+        -0x36t
+        -0x7t
+        -0x32t
+        -0x44t
+        0x2dt
+        0x28t
+        -0x34t
+        0x64t
+        0x60t
+        -0x3et
+        -0x1et
+        -0x74t
+        0x59t
+        0x30t
+        0x28t
+        0x7et
+        -0x1bt
+        0x51t
+        -0x7dt
+        0x60t
+        -0x1at
+        0x52t
+        0x53t
+        -0x1et
+        0x39t
+        -0xdt
+        0x14t
+        0x5ct
+        0x30t
+        -0x54t
+        -0x2at
+        -0x15t
+        0x3ct
+        -0xct
+        -0xdt
+        0x37t
+        0x25t
+        -0x25t
+        0x75t
+        0x52t
+        0x2et
+        -0x58t
+        -0x7ct
+        0x34t
+        -0x27t
+        -0x5t
+        -0x79t
+        0xet
+        0x5dt
+        0x3ct
+        0x7ft
+        -0x6bt
+        -0x63t
+        0x0t
+        -0x55t
+        -0x51t
+        0xbt
+        -0x80t
+        0x71t
+        -0x80t
+        0x8t
+        0x5ct
+        -0x52t
+        -0x53t
+        0x65t
+        -0x22t
+        -0x16t
+        -0xet
+        0x68t
+        -0x73t
+        0x2ft
+        -0x72t
+        0x4at
+        -0x23t
+        0x5et
+        -0x6bt
+        0x79t
+        -0x33t
+        0x28t
+        -0x6t
+        -0x30t
+        -0x7et
+        -0x63t
+        -0x5ft
+        -0x7bt
+        -0xft
+        0x41t
+        -0x6bt
+        -0x2ct
+        -0x61t
+        -0x3et
+        0x2et
+        0x7t
+        0x56t
+        -0x5et
+        -0x75t
+        0xct
+        -0x80t
+        0x4t
+        0x41t
+        0x8t
+        0x12t
+        0x4ft
+        0x5et
+        0xct
+        -0x7ft
+        -0x80t
+        -0x24t
+        -0x6ft
+        0xat
+        0x5dt
+        0x1et
+        -0x28t
+        0x23t
+        -0x6t
+        -0x53t
+        -0x5ft
+        -0x34t
+        0x14t
+        0x20t
+        0x49t
+        -0x72t
+        -0x42t
+        0x44t
+        -0xft
+        -0x7ct
+        -0x66t
+        0x2et
+        -0x11t
+        0x51t
+        -0x4ft
+        -0xft
+        0x3ct
+        -0x2at
+        0xdt
+        0x69t
+        -0x25t
+        0x30t
+        0x25t
+        0x2bt
+        0x4et
+        0x23t
+        0x54t
+        0x27t
+        0x68t
+        0x59t
+        -0xct
+        0x34t
+        0x4ft
+        0x70t
+        0x2bt
+        -0x4t
+        -0x7dt
+        -0x2at
+        -0x6dt
+        0x10t
+        0x38t
+        0x1et
+        0x13t
+        -0xat
+        0x2ft
+        -0x50t
+        0x59t
+        -0x57t
+        0x1dt
+        -0x39t
+        0x75t
+        0x74t
+        -0x64t
+        0x62t
+        0x31t
+        0x75t
+        0x4et
+        0x4at
+        -0x34t
+        0x0t
+        0x0t
+        0x6bt
+        -0x2ct
+        0x4at
+        -0x33t
+        -0xdt
+        -0x7t
+        -0x2at
+        0x38t
+        0x4at
+        -0x34t
+        -0x77t
+        -0x6ct
+        0x0t
+        0x0t
+        0x3t
+        0x2t
+        0x0t
+        0x0t
+        0x3t
+        0x0t
+        0x10t
+        0x0t
+        0x0t
+        0x3t
+        0x0t
+        0x10t
+        0x0t
+        0x0t
+        0x3t
+        0x0t
+        0x38t
+        0x6ct
+        0x44t
+        -0x7at
+        0x3t
+        0xct
+        0x14t
+        -0x44t
+        0x61t
+        0x1ct
+        0xat
+        0x28t
+        0x0t
+        0x0t
+        0x3t
+        0x3t
+        0x4ct
+        0x7ct
+        -0x4bt
+        -0x1t
+        -0x2t
+        0x0t
+        0x0t
+        0x3t
+        0x0t
+        0x0t
+        0x3t
+        0x0t
+        0x0t
+        0x3t
+        0x0t
+        0x0t
+        -0x40t
+        0x40t
+        -0x5t
+        0x1t
+        -0x3et
+        -0x5et
+        0x0t
+        0x30t
+        0x8t
+        0x0t
+        0x5ft
+        -0x25t
+        0x36t
+        -0x80t
+        -0x80t
+        0x50t
+        0x0t
+        0x0t
+        0x3t
+        0x0t
+        0x0t
+        0x3t
+        0x0t
+        0x0t
+        0x3t
+        0x2t
+        0x2t
+        0x36t
+        -0x7ft
+        -0x51t
+        0x9t
+        0x41t
+        0x20t
+        0x5t
+        0xbt
+        0x4t
+        0x0t
+        0x8t
+        0x0t
+        0x7ft
+        -0x40t
+        0x0t
+        0x42t
+        -0x3et
+        0x3ft
+        -0x1at
+        0x6at
+        0x0t
+        0x22t
+        -0x7et
+        0x0t
+        0x20t
+        0x2t
+        0x0t
+        0x5t
+        -0x7ct
+        0xat
+        0x40t
+        0x4t
+        0x0t
+        0x3bt
+        0x64t
+        0x0t
+        0x40t
+        0x4t
+        0x0t
+        0x18t
+        0x2bt
+        -0x1t
+        -0x64t
+        0x0t
+        0x0t
+        0x8t
+        -0x3ft
+        0x42t
+        -0x6ct
+        -0x20t
+        0x0t
+        0x0t
+        0x46t
+        0x3t
+        -0x80t
+        0x8t
+        0x0t
+        -0x80t
+        0x0t
+        0x72t
+        -0x76t
+        0x7bt
+        -0x11t
+        -0x80t
+    .end array-data
+.end method
+
+.method public constructor <init>()V
+    .locals 0
+
+    .line 10
+    invoke-direct {p0}, Ljava/lang/Object;-><init>()V
+
+    return-void
+.end method
+
+.method static dv(Lmm/i;)Z
+    .locals 1
+
+    .line 15
+    iget-object v0, p0, Lmm/i;->x0:Lmm/i$f;
+
+    if-eqz v0, :cond_d
+
+    iget-object p0, p0, Lmm/i;->x0:Lmm/i$f;
+
+    iget p0, p0, Lmm/i$f;->f:I
+
+    const/4 v0, 0x7
+
+    if-ne p0, v0, :cond_d
+
+    const/4 p0, 0x1
+
+    goto :goto_e
+
+    :cond_d
+    const/4 p0, 0x0
+
+    :goto_e
+    return p0
+.end method
+
+.method public static in(Lmm/i;)V
+    .locals 3
+
+    .line 24
+    iget-object v0, p0, Lmm/i;->h0:Landroid/media/MediaFormat;
+
+    .line 25
+    if-eqz v0, :cond_28
+
+    invoke-static {p0}, Lmm/DV;->dv(Lmm/i;)Z
+
+    move-result p0
+
+    if-nez p0, :cond_b
+
+    goto :goto_28
+
+    .line 26
+    :cond_b
+    const-string p0, "profile"
+
+    const/4 v1, 0x2
+
+    invoke-virtual {v0, p0, v1}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
+
+    .line 27
+    const-string p0, "level"
+
+    invoke-virtual {v0, p0}, Landroid/media/MediaFormat;->removeKey(Ljava/lang/String;)V
+
+    .line 28
+    const-string p0, "color-standard"
+
+    const/4 v2, 0x6
+
+    invoke-virtual {v0, p0, v2}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
+
+    .line 29
+    const-string p0, "color-transfer"
+
+    const/4 v2, 0x7
+
+    invoke-virtual {v0, p0, v2}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
+
+    .line 30
+    const-string p0, "color-range"
+
+    invoke-virtual {v0, p0, v1}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
+
+    .line 31
+    return-void
+
+    .line 25
+    :cond_28
+    :goto_28
+    return-void
+.end method
+
+.method public static out(Lmm/i;Landroid/media/MediaFormat;)V
+    .locals 3
+
+    .line 34
+    invoke-static {p0}, Lmm/DV;->dv(Lmm/i;)Z
+
+    move-result v0
+
+    if-eqz v0, :cond_29
+
+    const-string v0, "video/hevc"
+
+    const-string v1, "mime"
+
+    invoke-virtual {p1, v1}, Landroid/media/MediaFormat;->getString(Ljava/lang/String;)Ljava/lang/String;
+
+    move-result-object v2
+
+    invoke-virtual {v0, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result v0
+
+    if-nez v0, :cond_15
+
+    goto :goto_29
+
+    .line 35
+    :cond_15
+    const-string v0, "video/dolby-vision"
+
+    invoke-virtual {p1, v1, v0}, Landroid/media/MediaFormat;->setString(Ljava/lang/String;Ljava/lang/String;)V
+
+    .line 36
+    const-string v0, "profile"
+
+    const/16 v1, 0x100
+
+    invoke-virtual {p1, v0, v1}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
+
+    .line 37
+    const-string v0, "level"
+
+    iget p0, p0, Lmm/i;->g1:I
+
+    invoke-virtual {p1, v0, p0}, Landroid/media/MediaFormat;->setInteger(Ljava/lang/String;I)V
+
+    .line 38
+    return-void
+
+    .line 34
+    :cond_29
+    :goto_29
+    return-void
+.end method
+
+.method public static sample(Lmm/i;ILjava/nio/ByteBuffer;Landroid/media/MediaCodec$BufferInfo;)Ljava/nio/ByteBuffer;
+    .locals 6
+
+    .line 41
+    invoke-static {p0}, Lmm/DV;->dv(Lmm/i;)Z
+
+    move-result v0
+
+    if-eqz v0, :cond_4c
+
+    iget p0, p0, Lmm/i;->z:I
+
+    if-ne p1, p0, :cond_4c
+
+    iget p0, p3, Landroid/media/MediaCodec$BufferInfo;->flags:I
+
+    and-int/lit8 p0, p0, 0x2
+
+    if-nez p0, :cond_4c
+
+    iget p0, p3, Landroid/media/MediaCodec$BufferInfo;->size:I
+
+    if-gtz p0, :cond_15
+
+    goto :goto_4c
+
+    .line 42
+    :cond_15
+    iget p0, p3, Landroid/media/MediaCodec$BufferInfo;->size:I
+
+    sget-object p1, Lmm/DV;->RPU:[B
+
+    array-length p1, p1
+
+    add-int/2addr p0, p1
+
+    invoke-static {p0}, Ljava/nio/ByteBuffer;->allocateDirect(I)Ljava/nio/ByteBuffer;
+
+    move-result-object p0
+
+    .line 43
+    invoke-virtual {p2}, Ljava/nio/ByteBuffer;->duplicate()Ljava/nio/ByteBuffer;
+
+    move-result-object p1
+
+    .line 44
+    iget p2, p3, Landroid/media/MediaCodec$BufferInfo;->offset:I
+
+    iget v0, p3, Landroid/media/MediaCodec$BufferInfo;->size:I
+
+    add-int/2addr p2, v0
+
+    invoke-virtual {p1, p2}, Ljava/nio/ByteBuffer;->limit(I)Ljava/nio/ByteBuffer;
+
+    move-result-object p2
+
+    iget v0, p3, Landroid/media/MediaCodec$BufferInfo;->offset:I
+
+    invoke-virtual {p2, v0}, Ljava/nio/ByteBuffer;->position(I)Ljava/nio/ByteBuffer;
+
+    .line 45
+    invoke-virtual {p0, p1}, Ljava/nio/ByteBuffer;->put(Ljava/nio/ByteBuffer;)Ljava/nio/ByteBuffer;
+
+    move-result-object p1
+
+    sget-object p2, Lmm/DV;->RPU:[B
+
+    invoke-virtual {p1, p2}, Ljava/nio/ByteBuffer;->put([B)Ljava/nio/ByteBuffer;
+
+    move-result-object p1
+
+    invoke-virtual {p1}, Ljava/nio/ByteBuffer;->flip()Ljava/nio/ByteBuffer;
+
+    .line 46
+    invoke-virtual {p0}, Ljava/nio/ByteBuffer;->limit()I
+
+    move-result v2
+
+    iget-wide v3, p3, Landroid/media/MediaCodec$BufferInfo;->presentationTimeUs:J
+
+    iget v5, p3, Landroid/media/MediaCodec$BufferInfo;->flags:I
+
+    const/4 v1, 0x0
+
+    move-object v0, p3
+
+    invoke-virtual/range {v0 .. v5}, Landroid/media/MediaCodec$BufferInfo;->set(IIJI)V
+
+    .line 47
+    return-object p0
+
+    .line 41
+    :cond_4c
+    :goto_4c
+    return-object p2
+.end method
+
+.method public static want(Lom/t0;)Z
+    .locals 2
+
+    .line 19
+    invoke-virtual {p0}, Lom/t0;->Se()Z
+
+    move-result v0
+
+    if-eqz v0, :cond_24
+
+    const-string v0, "com.oplus.feature.video.dv.support"
+
+    invoke-static {v0}, Lcom/oplus/camera/configure/CameraConfig;->getConfigBooleanValue(Ljava/lang/String;)Z
+
+    move-result v0
+
+    if-eqz v0, :cond_24
+
+    .line 20
+    invoke-virtual {p0}, Lom/t0;->Qa()I
+
+    move-result v0
+
+    const/16 v1, 0x1e
+
+    if-ne v0, v1, :cond_24
+
+    const-string v0, "video_size_1080p"
+
+    invoke-virtual {p0}, Lom/t0;->h7()Ljava/lang/String;
+
+    move-result-object p0
+
+    invoke-virtual {v0, p0}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+
+    move-result p0
+
+    if-eqz p0, :cond_24
+
+    const/4 p0, 0x1
+
+    goto :goto_25
+
+    :cond_24
+    const/4 p0, 0x0
+
+    .line 19
+    :goto_25
+    return p0
+.end method
diff --git a/classes7/mm/i$b.smali b/classes7/mm/i$b.smali
index 0bdd715ed7f6865bcec0c1412be06d719243b34c..dfc95c204637801b3b09e3431227638fe45db11b 100644
--- a/classes7/mm/i$b.smali
+++ b/classes7/mm/i$b.smali
@@ -2931,7 +2931,7 @@
 
     .line 20
     .line 21
-    if-gez v2, :cond_d5
+    if-gez v2, :cond_d8
 
     .line 22
     .line 23
@@ -2939,7 +2939,7 @@
 
     .line 24
     .line 25
-    if-eqz v0, :cond_d5
+    if-eqz v0, :cond_d8
 
     .line 26
     .line 27
@@ -3037,7 +3037,7 @@
     move-exception p0
 
     .line 73
-    goto/16 :goto_d7
+    goto/16 :goto_da
 
     .line 74
     .line 75
@@ -3046,10 +3046,10 @@
 
     .line 76
     .line 77
+    invoke-static {v0, p2}, Lmm/DV;->out(Lmm/i;Landroid/media/MediaFormat;)V
+
     iget-object v2, v0, Lmm/i;->Z:Landroid/media/MediaMuxer;
 
-    .line 78
-    .line 79
     invoke-virtual {v2, p2}, Landroid/media/MediaMuxer;->addTrack(Landroid/media/MediaFormat;)I
 
     .line 80
@@ -3082,7 +3082,7 @@
     const/4 v3, 0x0
 
     .line 95
-    if-eqz v2, :cond_79
+    if-eqz v2, :cond_7c
 
     .line 96
     .line 97
@@ -3102,7 +3102,7 @@
     move-result v0
 
     .line 105
-    if-eqz v0, :cond_74
+    if-eqz v0, :cond_77
 
     .line 106
     .line 107
@@ -3118,30 +3118,30 @@
     move-result v0
 
     .line 113
-    if-eqz v0, :cond_74
+    if-eqz v0, :cond_77
 
     .line 114
     .line 115
     move v0, v1
 
     .line 116
-    goto :goto_75
+    goto :goto_78
 
     .line 117
-    :cond_74
+    :cond_77
     move v0, v3
 
     .line 118
-    :goto_75
+    :goto_78
     invoke-virtual {v2, v0}, Ljava/util/concurrent/atomic/AtomicBoolean;->set(Z)V
 
     .line 119
     .line 120
     .line 121
-    goto :goto_92
+    goto :goto_95
 
     .line 122
-    :cond_79
+    :cond_7c
     iget-object v2, v0, Lmm/i;->s1:Ljava/util/concurrent/atomic/AtomicBoolean;
 
     .line 123
@@ -3158,7 +3158,7 @@
     move-result v0
 
     .line 130
-    if-nez v0, :cond_8e
+    if-nez v0, :cond_91
 
     .line 131
     .line 132
@@ -3174,32 +3174,32 @@
     move-result v0
 
     .line 138
-    if-eqz v0, :cond_8c
+    if-eqz v0, :cond_8f
 
     .line 139
     .line 140
-    goto :goto_8e
+    goto :goto_91
 
     .line 141
-    :cond_8c
+    :cond_8f
     move v0, v3
 
     .line 142
-    goto :goto_8f
+    goto :goto_92
 
     .line 143
-    :cond_8e
-    :goto_8e
+    :cond_91
+    :goto_91
     move v0, v1
 
     .line 144
-    :goto_8f
+    :goto_92
     invoke-virtual {v2, v0}, Ljava/util/concurrent/atomic/AtomicBoolean;->set(Z)V
 
     .line 145
     .line 146
     .line 147
-    :goto_92
+    :goto_95
     iget-object v0, p0, Lmm/i$b;->a:Lmm/i;
 
     .line 148
@@ -3216,7 +3216,7 @@
     move-result v0
 
     .line 155
-    if-eqz v0, :cond_af
+    if-eqz v0, :cond_b2
 
     .line 156
     .line 157
@@ -3257,7 +3257,7 @@
     .line 174
     .line 175
     .line 176
-    :cond_af
+    :cond_b2
     sget-object v0, Lmm/i;->G1:Ljava/lang/String;
 
     .line 177
@@ -3328,17 +3328,17 @@
     .line 212
     .line 213
     .line 214
-    :cond_d5
+    :cond_d8
     monitor-exit p1
 
     .line 215
     return-void
 
     .line 216
-    :goto_d7
+    :goto_da
     monitor-exit p1
-    :try_end_d8
-    .catchall {:try_start_10 .. :try_end_d8} :catchall_47
+    :try_end_db
+    .catchall {:try_start_10 .. :try_end_db} :catchall_47
 
     .line 217
     throw p0
diff --git a/classes7/mm/i.smali b/classes7/mm/i.smali
index ef541886cf56c516e3ece4418963b6c532fc8dbc..3402adf68e9a7bbd151d0663a950ff3ffecc52d1 100644
--- a/classes7/mm/i.smali
+++ b/classes7/mm/i.smali
@@ -12921,7 +12921,7 @@
     move-exception v0
 
     .line 307
-    goto/16 :goto_2f1
+    goto/16 :goto_2f5
 
     .line 308
     .line 309
@@ -13039,7 +13039,7 @@
 
     .line 358
     .line 359
-    if-eqz v1, :cond_2c7
+    if-eqz v1, :cond_2cb
 
     .line 360
     .line 361
@@ -13047,7 +13047,7 @@
 
     .line 362
     .line 363
-    if-eqz v1, :cond_2c7
+    if-eqz v1, :cond_2cb
 
     .line 364
     .line 365
@@ -13673,17 +13673,21 @@
 
     .line 655
     .line 656
+    invoke-static {v7, v8, v0, v1}, Lmm/DV;->sample(Lmm/i;ILjava/nio/ByteBuffer;Landroid/media/MediaCodec$BufferInfo;)Ljava/nio/ByteBuffer;
+
+    move-result-object v0
+
     invoke-virtual {v2, v8, v0, v1}, Landroid/media/MediaMuxer;->writeSampleData(ILjava/nio/ByteBuffer;Landroid/media/MediaCodec$BufferInfo;)V
-    :try_end_292
-    .catch Ljava/lang/IllegalStateException; {:try_start_28d .. :try_end_292} :catch_293
+    :try_end_296
+    .catch Ljava/lang/IllegalStateException; {:try_start_28d .. :try_end_296} :catch_297
 
     .line 657
     .line 658
     .line 659
-    goto :goto_2b4
+    goto :goto_2b8
 
     .line 660
-    :catch_293
+    :catch_297
     move-exception v0
 
     .line 661
@@ -13747,7 +13751,7 @@
     .line 691
     .line 692
     .line 693
-    :goto_2b4
+    :goto_2b8
     iget v0, v9, Landroid/media/MediaCodec$BufferInfo;->flags:I
 
     .line 694
@@ -13755,7 +13759,7 @@
     const/4 v1, 0x1
 
     .line 696
-    if-ne v1, v0, :cond_2da
+    if-ne v1, v0, :cond_2de
 
     .line 697
     .line 698
@@ -13763,7 +13767,7 @@
 
     .line 699
     .line 700
-    if-ne v0, v8, :cond_2da
+    if-ne v0, v8, :cond_2de
 
     .line 701
     .line 702
@@ -13775,7 +13779,7 @@
 
     .line 705
     .line 706
-    if-nez v0, :cond_2da
+    if-nez v0, :cond_2de
 
     .line 707
     .line 708
@@ -13784,15 +13788,15 @@
     .line 709
     .line 710
     .line 711
-    goto :goto_2da
+    goto :goto_2de
 
     .line 712
-    :cond_2c7
+    :cond_2cb
     iget-object v0, v7, Lmm/i;->Z:Landroid/media/MediaMuxer;
 
     .line 713
     .line 714
-    if-eqz v0, :cond_2da
+    if-eqz v0, :cond_2de
 
     .line 715
     .line 716
@@ -13825,8 +13829,8 @@
     .line 729
     .line 730
     .line 731
-    :cond_2da
-    :goto_2da
+    :cond_2de
+    :goto_2de
     invoke-static {}, Ljava/lang/System;->currentTimeMillis()J
 
     .line 732
@@ -13846,7 +13850,7 @@
 
     .line 739
     .line 740
-    if-lez v2, :cond_2f0
+    if-lez v2, :cond_2f4
 
     .line 741
     .line 742
@@ -13871,15 +13875,15 @@
     .line 751
     .line 752
     .line 753
-    :cond_2f0
+    :cond_2f4
     return-void
 
     .line 754
-    :goto_2f1
-    :try_start_2f1
+    :goto_2f5
+    :try_start_2f5
     monitor-exit v1
-    :try_end_2f2
-    .catchall {:try_start_2f1 .. :try_end_2f2} :catchall_131
+    :try_end_2f6
+    .catchall {:try_start_2f5 .. :try_end_2f6} :catchall_131
 
     .line 755
     throw v0
@@ -18113,7 +18117,7 @@
 
     .line 14
     :cond_d
-    const-string p0, "video/dolby-vision"
+    const-string p0, "video/hevc"
 
     .line 15
     .line 16
@@ -18885,6 +18889,8 @@
     .line 227
     .line 228
     :cond_e3
+    invoke-static {p0}, Lmm/DV;->in(Lmm/i;)V
+
     iget-object p0, p0, Lmm/i;->h0:Landroid/media/MediaFormat;
 
     .line 229
diff --git a/classes7/om/t0$b.smali b/classes7/om/t0$b.smali
index b737e9d3d2c1d8d8ad48ef6eddc157cb392f6642..ca2502e16abc8f0248f60197b91a69339f65f66f 100644
--- a/classes7/om/t0$b.smali
+++ b/classes7/om/t0$b.smali
@@ -1845,23 +1845,15 @@
     const/4 v8, 0x0
 
     .line 338
-    if-eqz v9, :cond_197
+    if-eqz v9, :cond_195
 
     .line 339
     .line 340
-    const-string v7, "com.oplus.feature.video.hlg10.support"
+    invoke-static {v2}, Lmm/DV;->want(Lom/t0;)Z
 
-    .line 341
-    .line 342
-    invoke-static {v7}, Lcom/oplus/camera/configure/CameraConfig;->getConfigBooleanValue(Ljava/lang/String;)Z
-
-    .line 343
-    .line 344
-    .line 345
     move-result v7
 
-    .line 346
-    if-eqz v7, :cond_17d
+    if-nez v7, :cond_17b
 
     .line 347
     .line 348
@@ -1881,7 +1873,7 @@
     move-result v7
 
     .line 356
-    if-eqz v7, :cond_171
+    if-eqz v7, :cond_16f
 
     .line 357
     .line 358
@@ -1906,10 +1898,10 @@
     .line 367
     .line 368
     .line 369
-    goto :goto_1d4
+    goto :goto_1d2
 
     .line 370
-    :cond_171
+    :cond_16f
     const/4 v8, 0x2
 
     .line 371
@@ -1931,10 +1923,10 @@
     .line 379
     .line 380
     .line 381
-    goto :goto_1d4
+    goto :goto_1d2
 
     .line 382
-    :cond_17d
+    :cond_17b
     const/4 v7, 0x7
 
     .line 383
@@ -1984,10 +1976,10 @@
     .line 405
     .line 406
     .line 407
-    goto :goto_1d4
+    goto :goto_1d2
 
     .line 408
-    :cond_197
+    :cond_195
     invoke-virtual {v2}, Lom/t0;->Jk()Z
 
     .line 409
@@ -1996,7 +1988,7 @@
     move-result v8
 
     .line 412
-    if-nez v8, :cond_1c7
+    if-nez v8, :cond_1c5
 
     .line 413
     .line 414
@@ -2008,14 +2000,14 @@
     move-result v8
 
     .line 418
-    if-eqz v8, :cond_1a4
+    if-eqz v8, :cond_1a2
 
     .line 419
     .line 420
-    goto :goto_1c7
+    goto :goto_1c5
 
     .line 421
-    :cond_1a4
+    :cond_1a2
     const-string v8, "H265"
 
     .line 422
@@ -2028,7 +2020,7 @@
     move-result v8
 
     .line 427
-    if-eqz v8, :cond_1af
+    if-eqz v8, :cond_1ad
 
     .line 428
     .line 429
@@ -2036,10 +2028,10 @@
 
     .line 430
     .line 431
-    goto :goto_1d4
+    goto :goto_1d2
 
     .line 432
-    :cond_1af
+    :cond_1ad
     const-string v8, "H264"
 
     .line 433
@@ -2052,7 +2044,7 @@
     move-result v7
 
     .line 438
-    if-eqz v7, :cond_1d4
+    if-eqz v7, :cond_1d2
 
     .line 439
     .line 440
@@ -2085,11 +2077,11 @@
     .line 453
     .line 454
     .line 455
-    goto :goto_1d4
+    goto :goto_1d2
 
     .line 456
-    :cond_1c7
-    :goto_1c7
+    :cond_1c5
+    :goto_1c5
     iput v13, v3, Landroid/media/CamcorderProfile;->videoCodec:I
 
     .line 457
@@ -2115,8 +2107,8 @@
     .line 467
     .line 468
     .line 469
-    :cond_1d4
-    :goto_1d4
+    :cond_1d2
+    :goto_1d2
     invoke-static {}, Lcom/oplus/camera/common/gl/a;->b()Z
 
     .line 470
@@ -2157,7 +2149,7 @@
 
     .line 488
     .line 489
-    if-nez v7, :cond_2f4
+    if-nez v7, :cond_2f2
 
     .line 490
     .line 491
@@ -2169,15 +2161,15 @@
     move-result v7
 
     .line 495
-    if-eqz v7, :cond_1f2
+    if-eqz v7, :cond_1f0
 
     .line 496
     .line 497
-    goto/16 :goto_2f4
+    goto/16 :goto_2f2
 
     .line 498
     .line 499
-    :cond_1f2
+    :cond_1f0
     invoke-virtual {v2}, Lcom/oplus/camera/module/BaseMode;->uc()Z
 
     .line 500
@@ -2190,11 +2182,11 @@
 
     .line 504
     .line 505
-    if-eqz v7, :cond_20c
+    if-eqz v7, :cond_20a
 
     .line 506
     .line 507
-    if-eqz v4, :cond_20c
+    if-eqz v4, :cond_20a
 
     .line 508
     .line 509
@@ -2206,7 +2198,7 @@
     move-result v4
 
     .line 513
-    if-eqz v4, :cond_20c
+    if-eqz v4, :cond_20a
 
     .line 514
     .line 515
@@ -2227,10 +2219,10 @@
     .line 522
     .line 523
     .line 524
-    goto :goto_218
+    goto :goto_216
 
     .line 525
-    :cond_20c
+    :cond_20a
     iget-object v4, v2, Lom/t0;->N2:Landroid/media/CamcorderProfile;
 
     .line 526
@@ -2255,7 +2247,7 @@
     .line 535
     .line 536
     .line 537
-    :goto_218
+    :goto_216
     iget-object v4, v2, Lom/t0;->N2:Landroid/media/CamcorderProfile;
 
     .line 538
@@ -2360,7 +2352,7 @@
     move-result v4
 
     .line 591
-    if-eqz v4, :cond_268
+    if-eqz v4, :cond_266
 
     .line 592
     .line 593
@@ -2392,7 +2384,7 @@
     move-result-object v6
 
     .line 607
-    if-eqz v6, :cond_292
+    if-eqz v6, :cond_290
 
     .line 608
     .line 609
@@ -2404,17 +2396,17 @@
     move-result v6
 
     .line 613
-    if-eqz v6, :cond_292
+    if-eqz v6, :cond_290
 
     .line 614
     .line 615
     move v4, v6
 
     .line 616
-    goto :goto_292
+    goto :goto_290
 
     .line 617
-    :cond_268
+    :cond_266
     invoke-virtual {v2}, Lom/t0;->Pj()Z
 
     .line 618
@@ -2423,7 +2415,7 @@
     move-result v4
 
     .line 621
-    if-eqz v4, :cond_285
+    if-eqz v4, :cond_283
 
     .line 622
     .line 623
@@ -2473,10 +2465,10 @@
     .line 643
     .line 644
     .line 645
-    goto :goto_292
+    goto :goto_290
 
     .line 646
-    :cond_285
+    :cond_283
     invoke-virtual {v2}, Lom/t0;->Zj()Z
 
     .line 647
@@ -2485,7 +2477,7 @@
     move-result v4
 
     .line 650
-    if-eqz v4, :cond_290
+    if-eqz v4, :cond_28e
 
     .line 651
     .line 652
@@ -2499,16 +2491,16 @@
     div-int/2addr v4, v6
 
     .line 656
-    goto :goto_292
+    goto :goto_290
 
     .line 657
-    :cond_290
+    :cond_28e
     iget v4, v3, Landroid/media/CamcorderProfile;->videoBitRate:I
 
     .line 658
     .line 659
-    :cond_292
-    :goto_292
+    :cond_290
+    :goto_290
     invoke-static {v4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
 
     .line 660
@@ -2526,7 +2518,7 @@
 
     .line 667
     .line 668
-    if-eqz v4, :cond_2ef
+    if-eqz v4, :cond_2ed
 
     .line 669
     .line 670
@@ -2550,7 +2542,7 @@
     move-result v4
 
     .line 680
-    if-nez v4, :cond_2bd
+    if-nez v4, :cond_2bb
 
     .line 681
     .line 682
@@ -2574,14 +2566,14 @@
     move-result v4
 
     .line 692
-    if-eqz v4, :cond_2b6
+    if-eqz v4, :cond_2b4
 
     .line 693
     .line 694
-    goto :goto_2bd
+    goto :goto_2bb
 
     .line 695
-    :cond_2b6
+    :cond_2b4
     iget-object v4, v2, Lom/t0;->N2:Landroid/media/CamcorderProfile;
 
     .line 696
@@ -2594,11 +2586,11 @@
 
     .line 700
     .line 701
-    goto :goto_2c7
+    goto :goto_2c5
 
     .line 702
-    :cond_2bd
-    :goto_2bd
+    :cond_2bb
+    :goto_2bb
     iget-object v4, v2, Lom/t0;->N2:Landroid/media/CamcorderProfile;
 
     .line 703
@@ -2621,7 +2613,7 @@
     div-int/2addr v4, v7
 
     .line 712
-    :goto_2c7
+    :goto_2c5
     sget-object v7, Lmm/a0$g;->K:Lcom/adobe/internal/xmp/impl/i;
 
     .line 713
@@ -2698,7 +2690,7 @@
     .line 750
     .line 751
     .line 752
-    :cond_2ef
+    :cond_2ed
     const/4 v6, 0x2
 
     .line 753
@@ -2706,12 +2698,12 @@
 
     .line 754
     .line 755
-    goto/16 :goto_3a6
+    goto/16 :goto_3a4
 
     .line 756
     .line 757
-    :cond_2f4
-    :goto_2f4
+    :cond_2f2
+    :goto_2f2
     invoke-virtual {v2}, Lcom/oplus/camera/module/BaseMode;->ed()Z
 
     .line 758
@@ -2720,7 +2712,7 @@
     move-result v4
 
     .line 761
-    if-nez v4, :cond_302
+    if-nez v4, :cond_300
 
     .line 762
     .line 763
@@ -2736,11 +2728,11 @@
     move-result v4
 
     .line 769
-    if-eqz v4, :cond_30e
+    if-eqz v4, :cond_30c
 
     .line 770
     .line 771
-    :cond_302
+    :cond_300
     invoke-virtual {v2}, Lcom/oplus/camera/module/BaseMode;->E9()Ljava/lang/String;
 
     .line 772
@@ -2757,26 +2749,26 @@
     move-result v4
 
     .line 779
-    if-eqz v4, :cond_30e
+    if-eqz v4, :cond_30c
 
     .line 780
     .line 781
     const/4 v4, 0x1
 
     .line 782
-    goto :goto_30f
+    goto :goto_30d
 
     .line 783
-    :cond_30e
+    :cond_30c
     const/4 v4, 0x0
 
     .line 784
-    :goto_30f
+    :goto_30d
     iget-object v6, v2, Lcom/oplus/camera/module/BaseMode;->M1:Lcom/oplus/camera/q6;
 
     .line 785
     .line 786
-    if-eqz v6, :cond_31b
+    if-eqz v6, :cond_319
 
     .line 787
     .line 788
@@ -2788,42 +2780,42 @@
     move-result v6
 
     .line 792
-    if-eqz v6, :cond_31b
+    if-eqz v6, :cond_319
 
     .line 793
     .line 794
     const/4 v6, 0x1
 
     .line 795
-    goto :goto_31c
+    goto :goto_31a
 
     .line 796
-    :cond_31b
+    :cond_319
     const/4 v6, 0x0
 
     .line 797
-    :goto_31c
-    if-nez v4, :cond_324
+    :goto_31a
+    if-nez v4, :cond_322
 
     .line 798
     .line 799
-    if-eqz v6, :cond_321
+    if-eqz v6, :cond_31f
 
     .line 800
     .line 801
-    goto :goto_324
+    goto :goto_322
 
     .line 802
-    :cond_321
+    :cond_31f
     const/16 v7, 0xb4
 
     .line 803
     .line 804
-    goto :goto_342
+    goto :goto_340
 
     .line 805
-    :cond_324
-    :goto_324
+    :cond_322
+    :goto_322
     invoke-virtual {v2}, Lom/t0;->fk()I
 
     .line 806
@@ -2840,22 +2832,22 @@
 
     .line 812
     .line 813
-    if-eqz v4, :cond_330
+    if-eqz v4, :cond_32e
 
     .line 814
     .line 815
-    if-ne v4, v7, :cond_332
+    if-ne v4, v7, :cond_330
 
     .line 816
     .line 817
-    :cond_330
+    :cond_32e
     const/4 v4, 0x2
 
     .line 818
-    goto :goto_33b
+    goto :goto_339
 
     .line 819
-    :cond_332
+    :cond_330
     const/4 v4, 0x1
 
     .line 820
@@ -2872,10 +2864,10 @@
     .line 825
     .line 826
     .line 827
-    goto :goto_342
+    goto :goto_340
 
     .line 828
-    :goto_33b
+    :goto_339
     invoke-static {v4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
 
     .line 829
@@ -2889,7 +2881,7 @@
     .line 833
     .line 834
     .line 835
-    :goto_342
+    :goto_340
     invoke-virtual {v2}, Lom/t0;->h4()I
 
     .line 836
@@ -2937,7 +2929,7 @@
     move-result v4
 
     .line 860
-    if-eqz v4, :cond_374
+    if-eqz v4, :cond_372
 
     .line 861
     .line 862
@@ -2965,7 +2957,7 @@
     move-result-object v6
 
     .line 874
-    if-eqz v6, :cond_372
+    if-eqz v6, :cond_370
 
     .line 875
     .line 876
@@ -2977,22 +2969,22 @@
     move-result v6
 
     .line 880
-    if-eqz v6, :cond_372
+    if-eqz v6, :cond_370
 
     .line 881
     .line 882
     move v4, v6
 
     .line 883
-    :cond_372
-    :goto_372
+    :cond_370
+    :goto_370
     const/4 v6, 0x2
 
     .line 884
-    goto :goto_39f
+    goto :goto_39d
 
     .line 885
-    :cond_374
+    :cond_372
     invoke-virtual {v2}, Lom/t0;->Pj()Z
 
     .line 886
@@ -3001,7 +2993,7 @@
     move-result v4
 
     .line 889
-    if-eqz v4, :cond_391
+    if-eqz v4, :cond_38f
 
     .line 890
     .line 891
@@ -3051,10 +3043,10 @@
     .line 911
     .line 912
     .line 913
-    goto :goto_372
+    goto :goto_370
 
     .line 914
-    :cond_391
+    :cond_38f
     invoke-virtual {v2}, Lom/t0;->Zj()Z
 
     .line 915
@@ -3063,7 +3055,7 @@
     move-result v4
 
     .line 918
-    if-eqz v4, :cond_39c
+    if-eqz v4, :cond_39a
 
     .line 919
     .line 920
@@ -3077,10 +3069,10 @@
     div-int/2addr v4, v6
 
     .line 924
-    goto :goto_39f
+    goto :goto_39d
 
     .line 925
-    :cond_39c
+    :cond_39a
     const/4 v6, 0x2
 
     .line 926
@@ -3088,7 +3080,7 @@
 
     .line 927
     .line 928
-    :goto_39f
+    :goto_39d
     invoke-static {v4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
 
     .line 929
@@ -3102,7 +3094,7 @@
     .line 933
     .line 934
     .line 935
-    :goto_3a6
+    :goto_3a4
     invoke-virtual {v2}, Lom/t0;->hl()Z
 
     .line 936
@@ -3118,7 +3110,7 @@
     const/4 v11, 0x3
 
     .line 942
-    if-eqz v4, :cond_3e3
+    if-eqz v4, :cond_3e1
 
     .line 943
     .line 944
@@ -3126,7 +3118,7 @@
 
     .line 945
     .line 946
-    if-eqz v4, :cond_3e3
+    if-eqz v4, :cond_3e1
 
     .line 947
     .line 948
@@ -3158,7 +3150,7 @@
     move-result v4
 
     .line 962
-    if-eqz v4, :cond_3e3
+    if-eqz v4, :cond_3e1
 
     .line 963
     .line 964
@@ -3194,7 +3186,7 @@
     .line 978
     .line 979
     .line 980
-    if-nez v5, :cond_3db
+    if-nez v5, :cond_3d9
 
     .line 981
     .line 982
@@ -3206,11 +3198,11 @@
     move-result v4
 
     .line 986
-    if-eqz v4, :cond_3e9
+    if-eqz v4, :cond_3e7
 
     .line 987
     .line 988
-    :cond_3db
+    :cond_3d9
     invoke-static {v11}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
 
     .line 989
@@ -3224,10 +3216,10 @@
     .line 993
     .line 994
     .line 995
-    goto :goto_3e9
+    goto :goto_3e7
 
     .line 996
-    :cond_3e3
+    :cond_3e1
     const/4 v4, 0x0
 
     .line 997
@@ -3240,8 +3232,8 @@
     .line 1000
     .line 1001
     .line 1002
-    :cond_3e9
-    :goto_3e9
+    :cond_3e7
+    :goto_3e7
     iget-boolean v4, v2, Lom/t0;->i3:Z
 
     .line 1003
@@ -3249,7 +3241,7 @@
     const/4 v5, -0x1
 
     .line 1005
-    if-eqz v4, :cond_40c
+    if-eqz v4, :cond_40a
 
     .line 1006
     .line 1007
@@ -3257,7 +3249,7 @@
 
     .line 1008
     .line 1009
-    if-eqz v4, :cond_40c
+    if-eqz v4, :cond_40a
 
     .line 1010
     .line 1011
@@ -3297,7 +3289,7 @@
     move-result v4
 
     .line 1029
-    if-eqz v4, :cond_415
+    if-eqz v4, :cond_413
 
     .line 1030
     .line 1031
@@ -3310,10 +3302,10 @@
 
     .line 1035
     .line 1036
-    goto :goto_415
+    goto :goto_413
 
     .line 1037
-    :cond_40c
+    :cond_40a
     iput v5, v3, Landroid/media/CamcorderProfile;->audioCodec:I
 
     .line 1038
@@ -3331,8 +3323,8 @@
     .line 1044
     .line 1045
     .line 1046
-    :cond_415
-    :goto_415
+    :cond_413
+    :goto_413
     invoke-static {}, Lcom/oplus/camera/util/Util;->N0()Z
 
     .line 1047
@@ -3389,7 +3381,7 @@
 
     .line 1073
     .line 1074
-    if-eqz v4, :cond_507
+    if-eqz v4, :cond_505
 
     .line 1075
     .line 1076
@@ -3401,7 +3393,7 @@
     move-result v4
 
     .line 1080
-    if-eqz v4, :cond_507
+    if-eqz v4, :cond_505
 
     .line 1081
     .line 1082
@@ -3413,7 +3405,7 @@
     move-result v4
 
     .line 1086
-    if-nez v4, :cond_4b5
+    if-nez v4, :cond_4b3
 
     .line 1087
     .line 1088
@@ -3425,7 +3417,7 @@
     move-result v4
 
     .line 1092
-    if-nez v4, :cond_4b5
+    if-nez v4, :cond_4b3
 
     .line 1093
     .line 1094
@@ -3437,7 +3429,7 @@
     move-result v4
 
     .line 1098
-    if-nez v4, :cond_4b5
+    if-nez v4, :cond_4b3
 
     .line 1099
     .line 1100
@@ -3461,14 +3453,14 @@
     move-result v4
 
     .line 1110
-    if-eqz v4, :cond_458
+    if-eqz v4, :cond_456
 
     .line 1111
     .line 1112
-    goto :goto_4b5
+    goto :goto_4b3
 
     .line 1113
-    :cond_458
+    :cond_456
     invoke-static {}, Lcom/oplus/camera/data/DataManager;->getInstance()Lcom/oplus/camera/data/DataManager;
 
     .line 1114
@@ -3485,7 +3477,7 @@
     move-result v5
 
     .line 1121
-    if-eqz v5, :cond_465
+    if-eqz v5, :cond_463
 
     .line 1122
     .line 1123
@@ -3493,15 +3485,15 @@
 
     .line 1124
     .line 1125
-    goto :goto_467
+    goto :goto_465
 
     .line 1126
-    :cond_465
+    :cond_463
     sget-object v5, Ln9/f;->n:Lcom/oplus/camera/data/DataKey;
 
     .line 1127
     .line 1128
-    :goto_467
+    :goto_465
     invoke-virtual {v4, v5, v9}, Lcom/oplus/camera/data/DataManager;->b(Lcom/oplus/camera/data/DataKey;Ljava/lang/Object;)Ljava/lang/Object;
 
     .line 1129
@@ -3607,22 +3599,22 @@
     move-result v5
 
     .line 1181
-    if-eqz v5, :cond_4a0
+    if-eqz v5, :cond_49e
 
     .line 1182
     .line 1183
     const/4 v5, 0x1
 
     .line 1184
-    goto :goto_4a1
+    goto :goto_49f
 
     .line 1185
-    :cond_4a0
+    :cond_49e
     const/4 v5, 0x0
 
     .line 1186
-    :goto_4a1
-    if-eqz v4, :cond_4a6
+    :goto_49f
+    if-eqz v4, :cond_4a4
 
     .line 1187
     .line 1188
@@ -3630,23 +3622,23 @@
 
     .line 1189
     .line 1190
-    goto :goto_4a7
+    goto :goto_4a5
 
     .line 1191
-    :cond_4a6
+    :cond_4a4
     move-object v6, v14
 
     .line 1192
-    :goto_4a7
+    :goto_4a5
     iput-object v6, v2, Lom/t0;->e3:Ljava/lang/String;
 
     .line 1193
     .line 1194
-    if-eqz v4, :cond_4b2
+    if-eqz v4, :cond_4b0
 
     .line 1195
     .line 1196
-    if-eqz v5, :cond_4af
+    if-eqz v5, :cond_4ad
 
     .line 1197
     .line 1198
@@ -3654,29 +3646,29 @@
 
     .line 1199
     .line 1200
-    :cond_4af
+    :cond_4ad
     iput-object v15, v2, Lom/t0;->d3:Ljava/lang/String;
 
     .line 1201
     .line 1202
-    goto :goto_4b7
+    goto :goto_4b5
 
     .line 1203
-    :cond_4b2
+    :cond_4b0
     iput-object v13, v2, Lom/t0;->d3:Ljava/lang/String;
 
     .line 1204
     .line 1205
-    goto :goto_4b7
+    goto :goto_4b5
 
     .line 1206
-    :cond_4b5
-    :goto_4b5
+    :cond_4b3
+    :goto_4b3
     iput-object v13, v2, Lom/t0;->d3:Ljava/lang/String;
 
     .line 1207
     .line 1208
-    :goto_4b7
+    :goto_4b5
     sget-object v4, Lmm/a0$g;->P:Lcom/adobe/internal/xmp/impl/i;
 
     .line 1209
@@ -3789,14 +3781,14 @@
     float-to-int v4, v4
 
     .line 1264
-    if-gtz v4, :cond_4f2
+    if-gtz v4, :cond_4f0
 
     .line 1265
     .line 1266
     const/4 v4, 0x1
 
     .line 1267
-    :cond_4f2
+    :cond_4f0
     new-instance v5, Lia/i;
 
     .line 1268
@@ -3832,11 +3824,11 @@
     .line 1284
     .line 1285
     .line 1286
-    goto/16 :goto_587
+    goto/16 :goto_585
 
     .line 1287
     .line 1288
-    :cond_507
+    :cond_505
     invoke-static {}, Lcom/oplus/camera/util/Util;->N0()Z
 
     .line 1289
@@ -3845,7 +3837,7 @@
     move-result v4
 
     .line 1292
-    if-eqz v4, :cond_587
+    if-eqz v4, :cond_585
 
     .line 1293
     .line 1294
@@ -3857,7 +3849,7 @@
     move-result v4
 
     .line 1298
-    if-eqz v4, :cond_587
+    if-eqz v4, :cond_585
 
     .line 1299
     .line 1300
@@ -3869,7 +3861,7 @@
     move-result v4
 
     .line 1304
-    if-nez v4, :cond_576
+    if-nez v4, :cond_574
 
     .line 1305
     .line 1306
@@ -3881,7 +3873,7 @@
     move-result v4
 
     .line 1310
-    if-nez v4, :cond_576
+    if-nez v4, :cond_574
 
     .line 1311
     .line 1312
@@ -3893,14 +3885,14 @@
     move-result v4
 
     .line 1316
-    if-eqz v4, :cond_526
+    if-eqz v4, :cond_524
 
     .line 1317
     .line 1318
-    goto :goto_576
+    goto :goto_574
 
     .line 1319
-    :cond_526
+    :cond_524
     invoke-static {}, Lcom/oplus/camera/data/DataManager;->getInstance()Lcom/oplus/camera/data/DataManager;
 
     .line 1320
@@ -3917,7 +3909,7 @@
     move-result v5
 
     .line 1327
-    if-eqz v5, :cond_533
+    if-eqz v5, :cond_531
 
     .line 1328
     .line 1329
@@ -3925,15 +3917,15 @@
 
     .line 1330
     .line 1331
-    goto :goto_535
+    goto :goto_533
 
     .line 1332
-    :cond_533
+    :cond_531
     sget-object v5, Ln9/f;->n:Lcom/oplus/camera/data/DataKey;
 
     .line 1333
     .line 1334
-    :goto_535
+    :goto_533
     invoke-virtual {v4, v5, v9}, Lcom/oplus/camera/data/DataManager;->b(Lcom/oplus/camera/data/DataKey;Ljava/lang/Object;)Ljava/lang/Object;
 
     .line 1335
@@ -3954,7 +3946,7 @@
     move-result v4
 
     .line 1344
-    if-eqz v4, :cond_573
+    if-eqz v4, :cond_571
 
     .line 1345
     .line 1346
@@ -4040,41 +4032,41 @@
 
     .line 1387
     .line 1388
-    if-gez v4, :cond_56e
+    if-gez v4, :cond_56c
 
     .line 1389
     .line 1390
-    goto :goto_570
+    goto :goto_56e
 
     .line 1391
-    :cond_56e
+    :cond_56c
     move-object/from16 v15, v18
 
     .line 1392
     .line 1393
-    :goto_570
+    :goto_56e
     iput-object v15, v2, Lom/t0;->d3:Ljava/lang/String;
 
     .line 1394
     .line 1395
-    goto :goto_578
+    goto :goto_576
 
     .line 1396
-    :cond_573
+    :cond_571
     iput-object v13, v2, Lom/t0;->d3:Ljava/lang/String;
 
     .line 1397
     .line 1398
-    goto :goto_578
+    goto :goto_576
 
     .line 1399
-    :cond_576
-    :goto_576
+    :cond_574
+    :goto_574
     iput-object v13, v2, Lom/t0;->d3:Ljava/lang/String;
 
     .line 1400
     .line 1401
-    :goto_578
+    :goto_576
     new-instance v4, Lgi/a;
 
     .line 1402
@@ -4102,8 +4094,8 @@
     .line 1414
     .line 1415
     .line 1416
-    :cond_587
-    :goto_587
+    :cond_585
+    :goto_585
     sget-object v4, Lcom/oplus/camera/zf$c;->y1:Lcom/oplus/camera/zf$b;
 
     .line 1417
@@ -4132,7 +4124,7 @@
     move-result v4
 
     .line 1430
-    if-nez v4, :cond_59f
+    if-nez v4, :cond_59d
 
     .line 1431
     .line 1432
@@ -4148,11 +4140,11 @@
     move-result v4
 
     .line 1438
-    if-eqz v4, :cond_5e2
+    if-eqz v4, :cond_5e0
 
     .line 1439
     .line 1440
-    :cond_59f
+    :cond_59d
     invoke-virtual {v2}, Lcom/oplus/camera/module/BaseMode;->r()Z
 
     .line 1441
@@ -4179,21 +4171,21 @@
     move-result v6
 
     .line 1452
-    if-eqz v6, :cond_5af
+    if-eqz v6, :cond_5ad
 
     .line 1453
     .line 1454
     const/4 v8, 0x2
 
     .line 1455
-    goto :goto_5b0
+    goto :goto_5ae
 
     .line 1456
-    :cond_5af
+    :cond_5ad
     move v8, v4
 
     .line 1457
-    :goto_5b0
+    :goto_5ae
     sget-object v4, Lmm/a0$g;->S:Lcom/adobe/internal/xmp/impl/i;
 
     .line 1458
@@ -4239,7 +4231,7 @@
     move-result v6
 
     .line 1480
-    if-nez v6, :cond_5da
+    if-nez v6, :cond_5d8
 
     .line 1481
     .line 1482
@@ -4267,26 +4259,26 @@
     move-result v5
 
     .line 1494
-    if-eqz v5, :cond_5d8
+    if-eqz v5, :cond_5d6
 
     .line 1495
     .line 1496
-    goto :goto_5da
+    goto :goto_5d8
 
     .line 1497
-    :cond_5d8
+    :cond_5d6
     const/4 v5, 0x0
 
     .line 1498
-    goto :goto_5db
+    goto :goto_5d9
 
     .line 1499
-    :cond_5da
-    :goto_5da
+    :cond_5d8
+    :goto_5d8
     const/4 v5, 0x1
 
     .line 1500
-    :goto_5db
+    :goto_5d9
     invoke-static {v5}, Ljava/lang/Boolean;->valueOf(Z)Ljava/lang/Boolean;
 
     .line 1501
@@ -4300,7 +4292,7 @@
     .line 1505
     .line 1506
     .line 1507
-    :cond_5e2
+    :cond_5e0
     sget-object v4, Lmm/a0$g;->k:Lcom/adobe/internal/xmp/impl/i;
 
     .line 1508
@@ -4419,7 +4411,7 @@
     move-result-object v4
 
     .line 1567
-    if-eqz v4, :cond_634
+    if-eqz v4, :cond_632
 
     .line 1568
     .line 1569
@@ -4463,7 +4455,7 @@
     .line 1587
     .line 1588
     .line 1589
-    :cond_634
+    :cond_632
     invoke-virtual {v2}, Lom/t0;->ik()Ljava/lang/Double;
 
     .line 1590
@@ -4472,7 +4464,7 @@
     move-result-object v4
 
     .line 1593
-    if-eqz v4, :cond_63f
+    if-eqz v4, :cond_63d
 
     .line 1594
     .line 1595
@@ -4484,14 +4476,14 @@
     move-result v4
 
     .line 1599
-    goto :goto_640
+    goto :goto_63e
 
     .line 1600
-    :cond_63f
+    :cond_63d
     const/4 v4, -0x1
 
     .line 1601
-    :goto_640
+    :goto_63e
     sget-object v5, Lmm/a0$g;->z:Lcom/adobe/internal/xmp/impl/i;
 
     .line 1602
@@ -4500,18 +4492,18 @@
 
     .line 1604
     .line 1605
-    if-eqz v6, :cond_647
+    if-eqz v6, :cond_645
 
     .line 1606
     .line 1607
-    goto :goto_648
+    goto :goto_646
 
     .line 1608
-    :cond_647
+    :cond_645
     const/4 v4, -0x1
 
     .line 1609
-    :goto_648
+    :goto_646
     invoke-static {v4}, Ljava/lang/Integer;->valueOf(I)Ljava/lang/Integer;
 
     .line 1610
@@ -4553,7 +4545,7 @@
     move-result v4
 
     .line 1630
-    if-lez v4, :cond_673
+    if-lez v4, :cond_671
 
     .line 1631
     .line 1632
@@ -4565,18 +4557,18 @@
 
     .line 1635
     .line 1636
-    if-lez v5, :cond_668
+    if-lez v5, :cond_666
 
     .line 1637
     .line 1638
-    if-le v4, v5, :cond_668
+    if-le v4, v5, :cond_666
 
     .line 1639
     .line 1640
     move v4, v5
 
     .line 1641
-    :cond_668
+    :cond_666
     new-instance v5, Lf9/d;
 
     .line 1642
@@ -4595,15 +4587,15 @@
     .line 1649
     .line 1650
     .line 1651
-    goto :goto_675
+    goto :goto_673
 
     .line 1652
-    :cond_673
+    :cond_671
     iget v4, v2, Lom/t0;->g3:I
 
     .line 1653
     .line 1654
-    :goto_675
+    :goto_673
     sget-object v5, Lmm/a0$g;->h:Lcom/adobe/internal/xmp/impl/i;
 
     .line 1655
@@ -4735,7 +4727,7 @@
     move-result v3
 
     .line 1723
-    if-nez v3, :cond_6e7
+    if-nez v3, :cond_6e5
 
     .line 1724
     .line 1725
@@ -4751,7 +4743,7 @@
     move-result-object v3
 
     .line 1731
-    if-nez v3, :cond_6c8
+    if-nez v3, :cond_6c6
 
     .line 1732
     .line 1733
@@ -4762,10 +4754,10 @@
     const/4 v4, 0x0
 
     .line 1736
-    goto :goto_6e7
+    goto :goto_6e5
 
     .line 1737
-    :cond_6c8
+    :cond_6c6
     invoke-static {v3, v4}, Lab/r;->k(Lab/n1;I)I
 
     .line 1738
@@ -4782,7 +4774,7 @@
     move-result v3
 
     .line 1745
-    if-nez v3, :cond_6d5
+    if-nez v3, :cond_6d3
 
     .line 1746
     .line 1747
@@ -4790,15 +4782,15 @@
 
     .line 1748
     .line 1749
-    goto :goto_6d7
+    goto :goto_6d5
 
     .line 1750
-    :cond_6d5
+    :cond_6d3
     const/16 v22, 0xb4
 
     .line 1751
     .line 1752
-    :goto_6d7
+    :goto_6d5
     add-int v5, v5, v22
 
     .line 1753
@@ -4831,8 +4823,8 @@
     move v4, v3
 
     .line 1768
-    :cond_6e7
-    :goto_6e7
+    :cond_6e5
+    :goto_6e5
     sget-object v3, Lmm/a0$g;->l:Lcom/adobe/internal/xmp/impl/i;
 
     .line 1769
@@ -4866,7 +4858,7 @@
 
     .line 1784
     .line 1785
-    if-eqz v4, :cond_71c
+    if-eqz v4, :cond_71a
 
     .line 1786
     .line 1787
@@ -4878,7 +4870,7 @@
     move-result v3
 
     .line 1791
-    if-eqz v3, :cond_70e
+    if-eqz v3, :cond_70c
 
     .line 1792
     .line 1793
@@ -4890,7 +4882,7 @@
     move-result v3
 
     .line 1797
-    if-eqz v3, :cond_70e
+    if-eqz v3, :cond_70c
 
     .line 1798
     .line 1799
@@ -4907,10 +4899,10 @@
     .line 1804
     .line 1805
     .line 1806
-    goto :goto_721
+    goto :goto_71f
 
     .line 1807
-    :cond_70e
+    :cond_70c
     sget-object v3, Lmm/a0$g;->o:Lcom/adobe/internal/xmp/impl/i;
 
     .line 1808
@@ -4936,10 +4928,10 @@
     .line 1818
     .line 1819
     .line 1820
-    goto :goto_721
+    goto :goto_71f
 
     .line 1821
-    :cond_71c
+    :cond_71a
     iget-object v3, v3, Lom/y0;->b:Ljava/lang/String;
 
     .line 1822
@@ -4949,7 +4941,7 @@
     .line 1824
     .line 1825
     .line 1826
-    :goto_721
+    :goto_71f
     invoke-virtual {v2}, Lom/t0;->ck()Ljava/lang/Double;
 
     .line 1827
@@ -4958,7 +4950,7 @@
     move-result-object v3
 
     .line 1830
-    if-eqz v3, :cond_748
+    if-eqz v3, :cond_746
 
     .line 1831
     .line 1832
@@ -4986,7 +4978,7 @@
 
     .line 1843
     .line 1844
-    if-lez v3, :cond_748
+    if-lez v3, :cond_746
 
     .line 1845
     .line 1846
@@ -5025,7 +5017,7 @@
     .line 1863
     .line 1864
     .line 1865
-    :cond_748
+    :cond_746
     sget-object v3, Lmm/a0$g;->F:Lcom/adobe/internal/xmp/impl/i;
 
     .line 1866
@@ -5126,7 +5118,7 @@
     move-result-object v6
 
     .line 1915
-    if-eqz v4, :cond_784
+    if-eqz v4, :cond_782
 
     .line 1916
     .line 1917
@@ -5138,21 +5130,21 @@
     move-result v4
 
     .line 1921
-    if-eqz v4, :cond_784
+    if-eqz v4, :cond_782
 
     .line 1922
     .line 1923
     move v7, v3
 
     .line 1924
-    goto :goto_785
+    goto :goto_783
 
     .line 1925
-    :cond_784
+    :cond_782
     move v7, v5
 
     .line 1926
-    :goto_785
+    :goto_783
     iput-boolean v7, v1, Lom/j;->f:Z
 
     .line 1927
@@ -5302,7 +5294,7 @@
     move-result v3
 
     .line 2001
-    if-eqz v3, :cond_834
+    if-eqz v3, :cond_832
 
     .line 2002
     .line 2003
@@ -5502,7 +5494,7 @@
 
     .line 2100
     .line 2101
-    :cond_834
+    :cond_832
     iget-object v1, v2, Lom/t0;->x3:Ljava/util/concurrent/atomic/AtomicBoolean;
 
     .line 2102
diff --git a/classes7/om/t0.smali b/classes7/om/t0.smali
index d63a58c01b035f49a2934a89695aa35e416dfc7d..f429b9c5e23b02a8bfda5ad1a8331f3e5ba5b9e4 100644
--- a/classes7/om/t0.smali
+++ b/classes7/om/t0.smali
@@ -8747,91 +8747,29 @@
 .method public final Ik()Z
     .locals 1
 
-    .line 1
     invoke-virtual {p0}, Lcom/oplus/camera/module/BaseMode;->p2()Z
 
-    .line 2
-    .line 3
-    .line 4
     move-result v0
 
-    .line 5
-    if-nez v0, :cond_e
+    if-nez v0, :cond_11
 
-    .line 6
-    .line 7
     invoke-virtual {p0}, Lcom/oplus/camera/module/BaseMode;->We()Z
 
-    .line 8
-    .line 9
-    .line 10
-    move-result p0
+    move-result v0
 
-    .line 11
-    if-eqz p0, :cond_e
+    if-nez v0, :cond_10
 
-    .line 12
-    .line 13
-    const/4 p0, 0x1
+    invoke-static {p0}, Lmm/DV;->want(Lom/t0;)Z
 
-    .line 14
-    goto :goto_f
+    move-result v0
 
-    .line 15
-    :cond_e
-    const/4 p0, 0x0
+    :cond_10
+    return v0
 
-    .line 16
-    :goto_f
-    return p0
-    .line 17
-    .line 18
-    .line 19
-    .line 20
-    .line 21
-    .line 22
-    .line 23
-    .line 24
-    .line 25
-    .line 26
-    .line 27
-    .line 28
-    .line 29
-    .line 30
-    .line 31
-    .line 32
-    .line 33
-    .line 34
-    .line 35
-    .line 36
-    .line 37
-    .line 38
-    .line 39
-    .line 40
-    .line 41
-    .line 42
-    .line 43
-    .line 44
-    .line 45
-    .line 46
-    .line 47
-    .line 48
-    .line 49
-    .line 50
-    .line 51
-    .line 52
-    .line 53
-    .line 54
-    .line 55
-    .line 56
-    .line 57
-    .line 58
-    .line 59
-    .line 60
-    .line 61
-    .line 62
-    .line 63
-    .line 64
+    :cond_11
+    const/4 v0, 0x0
+
+    return v0
 .end method
 
 .method public Jf(I)Z
